src/test/rpc/TransactionEntry_test.cpp

```diff
@@ -244,10 +265,13 @@ class TransactionEntry_test : public beast::unit_test::suite
     "SigningPubKey" : "0324CAAFA2212D2AEAB9D42D481535614AED486293E1FB1380FF070C3DD7FB4264",
     "TransactionType" : "AccountSet",
     "TxnSignature" : "3044022007B35E3B99460534FF6BC3A66FBBA03591C355CC38E38588968E87CCD01BE229022071A443026DE45041B55ABB1CC76812A87EA701E475BBB7E165513B4B242D3474",
-    "hash" : "F4E9DF90D829A9E8B423FF68C34413E240D8D8BB0EFD080DF08114ED398E2506"
```

Comment:
I hate to see the hash check just be thrown away. Add a `hash` parameter to `check_tx`, and then check for it in the appropriate place depending on API version.

Reply:
There already is `txhash` parameter in `check_tx` and it is exactly what you are looking for i.e. `env.tx()->getTransactionID()`. However, I do agree that throwing these hash values is bad, so I added them back to verify the exact transaction hashes they were originally referring to, but with a code structure that's more natural for this particular check. Also, in the same spirit I added ledger hashes.